### ISSUE TITLE
Test summary during watch mode run

### ISSUE THREAD

**Issue by sethreidnz** [vitest-dev/vitest#3732]
### Clear and concise description of the problem

The proposal is to show a live-updating summary of the current test run instead of the list of tests that have ran and are running. This would provide a consistent UI that the user can see the progress of the run toward finishing while also outputting the error messages from failed tests as fast as possible. I believe this will both improve workflow of developers and improve the perceived performance of Vitest.

### Suggested solution

Instead of showing the list of all tests suites as they are ran as is the current UI, Vitest could instead show a live-updating summary of the current run of tests. This would look somewhat similar to the Jest watch UI and could show things such as:

- Total tests run 
- Total failures
- Total passes
- Total tests to run

While making this change it will also be important to ensure that:

- The UI is interruptible and responsive so that the user can pause execution and change things such as filename pattern
- Test failures are printed above the summary as soon as they happen so that the user can scroll up and see the first failed test as soon as possible when running multiple test suites.
- Test console output is also included in the correct order along with the test failures so that the messages can be read for debugging

### Alternative

I'm aware there is Vitest UI which I am yet to properly check out. On my current project I have had issues getting it going so haven't had a chance to look at it. Although it may improve on the current watch experience it would still be great to improve the default cli experience.

### Additional context

I would love to contribute to this and if you have other better ideas or plans that I've missed then let me know and I could help! I've just migrated a project onto Vite that has more than 10k tests and so far I'm really enjoying Vite!

### Validations

- [X] Follow our [Code of Conduct](https://github.com/vitest-dev/vitest/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guidelines](https://github.com/vitest-dev/vitest/blob/main/CONTRIBUTING.md).
- [X] Read the [docs](https://vitest.dev/guide/).
- [X] Check that there isn't already an issue that request the same feature to avoid creating a duplicate.

**Comment by sheremet-va** [comment]
Is it a duplicate of https://github.com/vitest-dev/vitest/issues/3549 ?

**Comment by sethreidnz** [comment]
> Is it a duplicate of #3549? (but better written)

Yes indeed it does look like it. Are you interested in contribution in the area?

**Comment by sheremet-va** [comment]
> > Is it a duplicate of #3549? (but better written)
> 
> Yes indeed it does look like it. Are you interested in contribution in the area?

Yes, PR is welcome.

**Comment by sethreidnz** [comment]
@sheremet-va I'll have a look at this! I have been given the go ahead to spend a little of my day job time on this so I will first have to upskill on how the project works and then hopefully be able to contribute something!

**Comment by sheremet-va** [comment]
> @sheremet-va I'll have a look at this! I have been given the go ahead to spend a little of my day job time on this so I will first have to upskill on how the project works and then hopefully be able to contribute something! Do you have any suggstions or guidance on learning about the codebase?

Renderer that prints test results is located here:

https://github.com/vitest-dev/vitest/blob/8dd5ea5ba10c2a99568ff18f6efff356824cb2cf/packages/vitest/src/node/reporters/renderers/listRenderer.ts#L89

Test summary is printed after all tests finished running, so you will need to figure out a way to print it alongside running tests:

https://github.com/vitest-dev/vitest/blob/8dd5ea5ba10c2a99568ff18f6efff356824cb2cf/packages/vitest/src/node/reporters/base.ts#L210

**Comment by AriPerkkio** [comment]
> #### Test summary during watch mode run

I think this should be based on `isTTY` instead of `--watch`/`--run`.

Even in `--run` mode this would be really useful, just like Jest does it.

**Comment by sethreidnz** [comment]
Hi yall! I really would like to work on this, however we've been finding since moving to vitest that our tests are running about 3x slower (from 25min with jest, to 1h 30m with vitest). We have been investigating all the possible options and what not but we may end up moving away from vitest (which I'm sad about) but that means it's hard for me to justify spending the time on this...

**Comment by sheremet-va** [comment]
> Hi yall! I really would like to work on this, however we've been finding since moving to vitest that our tests are running about 3x slower (from 25min with jest, to 1h 30m with vitest). We have been investigating all the possible options and what not but we may end up moving away from vitest (which I'm sad about) but that means it's hard for me to justify spending the time on this...

There is a performance regression in 0.34.0 that might cause this. I am currently working on this.

**Comment by hi-ogawa** [comment]
Just linking what Jest recently did to improve their summary output:
- https://github.com/jestjs/jest/pull/15008
